include/CLI/App.hpp

```diff
@@ -1720,44 +1778,58 @@ class App {
             app->_configure();
         }
     }
-    /// Internal function to run (App) callback, top down
+    /// Internal function to run (App) callback, bottom up
```

Comment:
This is a change in behavior,  the order of callbacks between the main app and child subcommands.  Previously it was a top down approach.  Now it is a bottom up approach.  There wasn't any tests that relied on this behavior but it is a change.   I would argue it makes more sense as a bottom up approach, to know that the main app callback only executes after everything else has completed.  But being there was a comment on the effect I wasn't entirely sure if there was a reason behind it or something that depended on that behavior.    I would say in most applications it would make no difference, but this is a change in behavior.

Reply:
Would a user be able to do something like:

```bash
git -C dir add -u 
```

That is, add would need to change to the `dir` directory. It's not too important, for this example or the existing code, since you don't have to run this in callbacks; you can just do a complete parse then inspect the results, which I assume is a bit more common.

Reply:
PS: I'm okay with the change, if I didn't say that clearly at the end (I didn't)